src/openai/lib/azure.py

```diff
@@ -58,11 +60,26 @@ def _build_request(
     ) -> httpx.Request:
         if options.url in _deployments_endpoints and is_mapping(options.json_data):
             model = options.json_data.get("model")
-            if model is not None and not "/deployments" in str(self.base_url):
+            if model is not None and "/deployments" not in str(self.base_url.path):
                 options.url = f"/deployments/{model}{options.url}"
 
         return super()._build_request(options, retries_taken=retries_taken)
 
+    @override
+    def _prepare_url(self, url: str) -> httpx.URL:
+        if not self._azure_deployment:
+            return super()._prepare_url(url)
+
+        deployment_segment = f"/deployments/{self._azure_deployment}"
+        if deployment_segment in str(self.base_url.path) and url not in _deployments_endpoints:
+            merge_url = httpx.URL(url)
+            if merge_url.is_relative_url:
+                base_path = self.base_url.path.split(deployment_segment)[0]
```

Comment:
Is this split performed on the last occurrence of deployment_segment?

Reply:
Done.